packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts

```diff
@@ -118,12 +148,17 @@ export class EcsTask implements events.IRuleTarget {
     this.taskDefinition = props.taskDefinition;
     this.taskCount = props.taskCount ?? 1;
     this.platformVersion = props.platformVersion;
+    this.propagateTags = props.propagateTags === true ? ecs.PropagatedTagSource.TASK_DEFINITION : undefined ;
```

Comment:
This is giving me pause. If CloudFormation expects something other than a true or false here, I think we're potentially setting ourselves up for breaking changes if they ever add more allowed values here.

Reply:
I can see how this could become problematic even though this is the only allowed value today. I can update this so it takes a value of type `ecs.PropagatedTagSource` and then validates that you passed the correct value with some helpful messaging. That way if a new source is ever added, the validation can be removed or modified accordingly.

Reply:
Updated!